openstack-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1004 @@
+        'Pool usage',
+        targets=[t.poolUsage],
+        description='The percent of pool capacity in use for Cinder.',
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    snapshots:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Snapshots',
+        targets=[t.snaphots],
+        description='The number of volume snapshots in Cinder.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    cinderAgents:
+      table.new('Agents')
+      + table.queryOptions.withTargets([t.cinderAgentState])
+      + table.queryOptions.withDatasource('prometheus', '${prometheus_datasource}')
+      + table.panelOptions.withDescription('Details for the agents for OpenStack Cinder.')
+      + table.fieldConfig.defaults.custom.withCellOptions('color-text')
+      + table.standardOptions.withOverridesMixin([
+        table.fieldOverride.byName.new('Admin state')
+        + table.fieldOverride.byName.withProperty('custom.displayMode', 'color-text')
+        + table.fieldOverride.byName.withPropertiesFromOptions(
+          table.standardOptions.withMappings(
+            {
+              type: 'value',
+              options: {
+                disabled: {
+                  color: 'red',
+                  index: 1,
+                  text: 'Disabled',
+                },
+                enabled: {
+                  color: 'green',
+                  index: 0,
+                  text: 'Enabled',
+                },
+              },
+            }
+          ),
+        ),
+      ])
+      + table.queryOptions.withTransformationsMixin([
+        {
+          id: 'joinByField',
+          options: {
+            byField: 'Time',
+            mode: 'outer',
+          },
+        },
+        {
+          id: 'organize',
+          options: {
+            excludeByName: {
+              Time: true,
+              Value: true,
+              __name__: true,
+              hostname: true,
+              job: true,
+            },
+            indexByName: {
+              Time: 6,
+              Value: 9,
+              __name__: 7,
+              adminState: 3,
+              hostname: 5,
+              instance: 1,
+              job: 8,
+              service: 0,
+              uuid: 4,
+              zone: 2,
+            },
+            renameByName: {
+              Time: '',
+              adminState: 'Admin state',
+              id: 'ID',
+              instance: 'Instance',
+              service: 'Service',
+              zone: 'Zone',
+              uuid: 'UUID',
+            },
+            includeByName: {},
+          },
+        },
+      ]),
+
+    imageCount:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Image count',
+        targets=[t.imageCount],
+        description='The number of images present in Glance.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
```

Comment:
Comparing to prototype missing size column, and some of the headers aren't renamed to be pretty looking.

Reply:
Looks like a copy-paste error with the wrong transformations, I'll get that fixed.

Reply:
This looks like the only table that isn't left justified. I'll leave that up to you to if you want to change. Good to resolve